src/core/QCheck.ml

```diff
@@ -209,6 +209,14 @@ module Gen = struct
     shuffle_a a st;
     Array.to_list a
 
+  let shuffle_w_l l st =
+    let sample (w, v) =
+      let fl_w = float_of_int w in
+      (float_bound_inclusive 1. st ** (1. /. fl_w), v)
+    in
+    let samples = List.map sample l in
```

Comment:
you could use `List.rev_map` since it's going to be sorted afterwards (and possibly sort it backwards to use `List.rev_map snd`).

Reply:
That's a good idea.